storage_backblaze.go

```diff
@@ -0,0 +1,193 @@
+		AccountID:      URL.User.Username(),
+		ApplicationKey: pw,
+	})
+	if err != nil {
+		return &StorageBackblaze{}, err
+	}
+
+	// Creating the bucket prefixes
+	bucketPrefix := strings.Split(URL.Path, "/")
+	if len(bucketPrefix) != 2 {
+		return &StorageBackblaze{}, ErrInvalidRepositoryURL
+	}
+
+	// TODO: Make me handsome
+	var bucket *backblaze.Bucket
+	bucket, err = cl.Bucket(bucketPrefix[1])
+	if err != nil || bucket == nil {
+		// Bucket probably doesn't exists
+		bucket, err = cl.CreateBucket(bucketPrefix[1], backblaze.AllPrivate)
+		if err != nil {
+			// Bucket exists but we don't have access on it
+			return &StorageBackblaze{}, err
+		}
+		if bucket == nil {
+			panic("bucket nil")
+		}
+		return &StorageBackblaze{
+			url:            URL,
+			chunkFile:      bucketPrefix[1] + "-chunks",
+			snapshotFile:   bucketPrefix[1] + "-snapshots",
+			repositoryFile: bucketPrefix[1] + "-repository",
+			// bucketPrefix:   bucketPrefix[1],
+			bucket:    bucket,
+			backblaze: cl,
+		}, nil
+	}
+	return &StorageBackblaze{
+		url:            URL,
+		chunkFile:      bucketPrefix[1] + "-chunks",
+		snapshotFile:   bucketPrefix[1] + "-snapshots",
+		repositoryFile: bucketPrefix[1] + "-repository",
+		// bucketPrefix:   bucketPrefix[1],
+		bucket:    bucket,
+		backblaze: cl,
+	}, nil
+
+}
+
+// Location returns the type and location of the repository
+func (backend *StorageBackblaze) Location() string {
+	return backend.url.String()
+}
+
+// Close the backend
+func (backend *StorageBackblaze) Close() error {
+	return nil
+}
+
+// Protocols returns the Protocol Schemes supported by this backend
+func (backend *StorageBackblaze) Protocols() []string {
+	return []string{"backblaze"}
+}
+
+// Description returns a user-friendly description for this backend
+func (backend *StorageBackblaze) Description() string {
+	return "Backblaze Storage"
+}
+
+// AvailableSpace returns the free space on this backend
+func (backend *StorageBackblaze) AvailableSpace() (uint64, error) {
+	// Currently not supported
+	return 0, ErrAvailableSpaceUnknown
+}
+
+// LoadChunk loads a Chunk from backblaze
+func (backend *StorageBackblaze) LoadChunk(shasum string, part, totalParts uint) (*[]byte, error) {
+	fileName := shasum + "." + strconv.FormatUint(uint64(part), 10) + "_" + strconv.FormatUint(uint64(totalParts), 10)
+	_, obj, err := backend.bucket.DownloadFileByName(fileName)
+	if err != nil {
+		return nil, err
+	}
+	data, err := ioutil.ReadAll(obj)
+	return &data, err
+}
+
+// StoreChunk stores a single Chunk from backblaze
+func (backend *StorageBackblaze) StoreChunk(shasum string, part, totalParts uint, data *[]byte) (size uint64, err error) {
+	fileName := shasum + "." + strconv.FormatUint(uint64(part), 10) + "_" + strconv.FormatUint(uint64(totalParts), 10)
+
+	buf := bytes.NewBuffer(*data)
+	metadata := make(map[string]string)
+	i, err := backend.bucket.UploadFile(fileName, metadata, buf)
+	filesize := backblaze.File(*i)
```

Comment:
Changed variable name in cf841b4c5bc48da592ebe2f6bca9cf21b8013d0b

Checking the error returned by UploadFile(...) added here 1bbbcf7c1dad8cbe2f11cdc569bc5f5b89415a54